plugins/org.wso2.developerstudio.eclipse.esb.docker/src/org/wso2/developerstudio/eclipse/esb/docker/wizard/ExportAndGenerateDockerImageWizard.java

```diff
@@ -0,0 +1,320 @@
+            }
+
+            // Copy the exported car file to the docker directory
+            File destFileName = new File(dockerDirPath, finalFileName);
+            IResource carbonArchive = ExportUtil.buildCAppProject(selectedProject);
+            FileUtils.copy(carbonArchive.getLocation().toFile(), destFileName);
+
+            // Create a docker model
+            MicroIntegratorDockerModel dockerModel = new MicroIntegratorDockerModel();
+            dockerModel.setName(detailsPage.getImageName());
+            dockerModel.setTag(detailsPage.getImageTag());
+            dockerModel.setCommandArg(DockerGenConstants.ImageParamDefaults.EI_START_COMMAND);
+            dockerModel.setPorts(DockerGenConstants.ImageParamDefaults.ports);
+            dockerModel.setServerHome(eiDistrubitionDestinationPath);
+
+            // Create and schedule a background job to generate the docker image
+            dockerJob = new GenerateDockerImageJob(dockerDirPath, eiDistributionSourcePath,
+                    eiDistrubitionDestinationPath, eiHomePath, deploymentPath, detailsPage.getExportPath(),
+                    destFileName, dockerModel);
+            dockerJob.schedule();
+
+        } catch (Exception e) {
+            log.error(ExportImageWizardConstants.ERROR_CREATING_CAR_FILE_MSG, e);
+            openMessageBox(getShell(), ExportImageWizardConstants.DIALOG_TITLE_TEXT,
+                    ExportImageWizardConstants.ERROR_CREATING_CAR_FILE_MSG + " For more details view the log.\n",
+                    SWT.ICON_ERROR);
+        }
+
+        setSessionProperty();
+
+        return true;
+    }
+
+    public static IProject getSelectedProject(Object obj) throws Exception {
+        if (obj == null) {
+            return null;
+        }
+
+        if (obj instanceof IResource) {
+            return ((IResource) obj).getProject();
+        } else if (obj instanceof IStructuredSelection) {
+            return getSelectedProject(((IStructuredSelection) obj).getFirstElement());
+        }
+
+        return null;
+    }
+
+    private void savePOM() throws Exception {
+        writeProperties();
+        parentPrj.setDependencies(new ArrayList<Dependency>(mainPage.getDependencyList().values()));
+        MavenUtils.saveMavenProject(parentPrj, pomFile);
+        pomFileRes.getProject().refreshLocal(IResource.DEPTH_INFINITE, new NullProgressMonitor());
+    }
+
+    private void writeProperties() {
+        Properties properties = parentPrj.getModel().getProperties();
+        identifyNonProjectProperties(properties);
+        for (Dependency dependency : mainPage.getDependencyList().values()) {
+            String artifactInfo = DistProjectUtils.getArtifactInfoAsString(dependency);
+
+            if (mainPage.getServerRoleList().containsKey(artifactInfo)) {
+                properties.put(artifactInfo, serverRoleList.get(artifactInfo));
+            } else {
+                properties.put(artifactInfo, "capp/ApplicationServer");
```

Comment:
Can we define constants? #241 and #245

Reply:
Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/800